### ISSUE TITLE
Template part: Area selection UI

### ISSUE THREAD

**Issue by jameskoster** [WordPress/gutenberg#29295]
When #29159 is merged it will be possible to update the `area` of the selected Template Part:

<img width="325" alt="Screenshot 2021-02-24 at 10 39 02" src="https://user-images.githubusercontent.com/846565/108988577-898ab500-768c-11eb-8dba-1a479251ae8d.png">

Since we have dedicated icons for these areas that will be surfaced to users in the Toolbar, List View, and the Inspector, it may be worth exploring designs that incorporate them into this selection flow in order to establish holistic continuity in the UI.

One option might be to use a segmented control instead of a dropdown:

<img src="https://user-images.githubusercontent.com/846565/108733393-60e8ab00-7526-11eb-9433-2dce688cfbe0.png" width="310">

Another option would be to stick with the select but enrich the dropdown:

<img width="310" alt="Screenshot 2021-02-24 at 11 04 13" src="https://user-images.githubusercontent.com/846565/108991465-0f5c2f80-7690-11eb-9a00-5d6ac4b62061.png">

Whichever solution we go with, we should be aware that this element will likely exist in other environments, e.g. #29222.

**Comment by mtias** [comment]
That last one seem like a lot. I can see it being useful when creating a new one from scratch, but to reassign it seems it should be simpler and assume context already (it's in "advanced" in the end).

**Comment by jameskoster** [comment]
I don't have strong feelings either way. I suppose the main thing to decide upon is whether we need to display the descriptions here.

If so, maybe there is a middle ground?

<img width="326" alt="Screenshot 2021-02-24 at 14 41 10" src="https://user-images.githubusercontent.com/846565/109016725-5b1dd180-76ae-11eb-8a96-b2ce9d8e8a95.png">

**Comment by jameskoster** [comment]
Let's try the segmented control / icon approach. We can see how adding the description as a tooltip feels:

<img width="465" alt="Screenshot 2021-03-02 at 11 04 43" src="https://user-images.githubusercontent.com/846565/109639523-32328c00-7b47-11eb-90da-0b7e34bbe8b3.png">

**Comment by annezazu** [comment]
This "Advanced" section came up as part of feedback from @paaljoachim for [the third FSE call for testing](https://make.wordpress.org/test/2021/03/09/fse-program-testing-call-3-create-a-fun-custom-404-page/#comment-1416). For context, part of this test includes creating a custom template part for the 404 page header:

> How do I give it a name. I looked into the Template Part block but could not see a way to rename it there.Ahh I had to open the Advanced panel. That is a weird place to put it. There I can rename it. Title, area and HTML element could be in an unique panel. The Advanced panel could be similar to what is seen in the Post Editor.

Right now, it does feel odd to have this living under Advanced settings when setting a name for the template part is likely the first action we should nudge a user to take so they can later tell the difference between template parts.

**Comment by paaljoachim** [comment]
Thanks for the ping @annezazu 

Tooltips
I am seeing that James it mentioning a kind of tooltip. I actually also mentioned this in the feedback I had for this third call for testing outreach. Having a ? in a circle or another method where one can hover a small icon to receive additional information will very much help. As there are options in the panels that are difficult to understand without any extra information.

Naming. 
Here is an example from Reusable blocks.

<img width="375" alt="Screen Shot 2021-03-17 at 23 46 04" src="https://user-images.githubusercontent.com/5323259/111548437-ff110f00-877a-11eb-9196-556ea795a939.png">

The question comes up. Should we have naming in an easier place to locate? As right now it seems like one is just adding on to the Advanced panel. Here is a suggestion.

![Header-name-FSE](https://user-images.githubusercontent.com/5323259/111549481-86ab4d80-877c-11eb-9fd7-eb80d7203d34.jpg)

**Comment by mtias** [comment]
> Right now, it does feel odd to have this living under Advanced settings when setting a name for the template part is likely the first action we should nudge a user to take so they can later tell the difference between template parts.

@annezazu that would be as part of creating a new template part, which has its own flow in a modal and a very prominent "name" field. Editing an existing name should not be as prominent.

**Comment by annezazu** [comment]
Makes sense for an existing template part but do want to point out that the current new template experience doesn't make it clear or easy to figure out where one can edit the name: 

https://user-images.githubusercontent.com/26996883/111723817-c4b27b00-8829-11eb-927b-4a75d11d0118.mov

I'll add this feedback to this issue for now: https://github.com/WordPress/gutenberg/issues/29950

**Comment by creativecoder** [comment]
@jameskoster [Here's a draft PR](https://github.com/WordPress/gutenberg/pull/30005) using the [segmented control / icon approach](https://github.com/WordPress/gutenberg/issues/29295#issuecomment-788825961) suggested above.

<img width="275" alt="image" src="https://user-images.githubusercontent.com/1699996/111732023-b15fd980-8842-11eb-8216-6ae1def06416.png">

It's roughed in for now, but gives a feel for using icon buttons with tooltips.

How can I find the icon for the General template option?

**Comment by jameskoster** [comment]
@annezazu I think that's a separate issue warranting dedicated exploration. It could be as simple as displaying a modal when the user elects to create a new template part. Or maybe even not allowing new template part creation via the block – that has always felt like a weird flow to me.

**Comment by jameskoster** [comment]
> How can I find the icon for the General template option?

It's [this one](https://github.com/WordPress/gutenberg/blob/trunk/packages/icons/src/library/layout.js) from the icons package :)

**Comment by jameskoster** [comment]
We can potentially consider reusing the popover pattern recently implemented for the document summary panel here. E.g.

https://user-images.githubusercontent.com/846565/177760933-30f8be28-4eff-44d8-a5ad-c7076b065e90.mp4

**Comment by paaljoachim** [comment]
That sounds like a good idea @jameskoster

**Comment by creativecoder** [comment]
I've made another attempt at this in https://github.com/WordPress/gutenberg/pull/45764 -- reviews welcome!

**Comment by jordesign** [comment]
Checkin in WP6.3 I'm now seeing the icon indicating the Template Area in the sidebar. @jameskoster @paaljoachim  do you feel like that is sufficient resolution, or should we leave this open for further refinement?

<img width="1701" alt="Screenshot 2023-09-08 at 12 59 32 pm" src="https://github.com/WordPress/gutenberg/assets/1842363/b478a8fb-bb79-4552-a80e-1e906fb55e64">

**Comment by jameskoster** [comment]
@jordesign this issue is about the appearance of the area selection UI. I reckon the UI might benefit from an update to bring it more inline with the add-template modal which was updated recently:

### Create template part
<img width="498" alt="Screenshot 2023-09-08 at 10 27 22" src="https://github.com/WordPress/gutenberg/assets/846565/a273401a-fa25-4cb1-8459-62901a794dab">

### Create template
<img width="863" alt="Screenshot 2023-09-08 at 10 29 08" src="https://github.com/WordPress/gutenberg/assets/846565/d175c478-4a33-402b-804f-f0f711bd7d11">
